lmfit/parameter.py

```diff
@@ -339,9 +339,10 @@ def add(self, name, value=None, vary=True, min=-inf, max=inf, expr=None,
 
         Parameters
         ----------
-        name : str
-            Name of parameter. Must match ``[a-z_][a-z0-9_]*`` and cannot
-            be a Python reserved word.
+        name : str or Parameter
+            Name of the parameter or a Parameter object. If the name is given, 
+            a new Parameter object will be created. The name must match 
+            ``[a-z_][a-z0-9_]*`` and cannot be a Python reserved word. 
```

Comment:
To me the way it's phrased now is a bit ambiguous; you say "If the name is given, ....." that will always be the case since `name` is a required argument. 

So to make it bit clearer (at least to me), how about something like:
```
If ``name`` refers to a Parameter object it will be added directly to the Parameters instance,
otherwise a new Parameter object with name ``string`` is created before adding it. In both 
cases,  ``name`` must match ``[a-z_][a-z0-9_]*`` and cannot be a Python reserved word.
```

Reply:
also, you left somewhere a "trailing-whitespace" in one of the lines, which made the `pre-commit` test fail - can you please fix that as well? Thanks!